### PR TITLE
Made HW5, p.1-3.

### PR DESCRIPTION
**Pull request by anastasiya-harbatsevich**
p1. Even Digits Sum
p.2 Generate Sort and Print Array
p.3 Number Generator

### CODE REVIEW

File: src/main/java/org/example/homework5/EvenDigitSum.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        int initialNumber = 2;
+        System.out.println(getEvenDigitSum(initialNumber));
+    }
+    public static int getEvenDigitSum(int number) {
+        if (number < 0) {
+            return -1;
+        }
+
+        int result = 0;
+        char[] stringNumber = Integer.toString(number).toCharArray();
```
Comment (alexshamrai) [inline src/main/java/org/example/homework5/EvenDigitSum.java]: Technically your solution is correct, but not efficient.
You can take a look at this one:
```
        int result = 0;
        while (number > 0) {
            int digit = number % 10;
            if (digit % 2 == 0) {
                sum += digit;
            }
            number /= 10;
        }
        return result;
```
It does not require an additional array and int to string manipulations

File: src/main/java/org/example/homework5/NumberGenerator.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public class NumberGenerator {
+    public static void main(String[] args) {
+        List<Integer> randomArray = generateRandomArray();
+        arraySum(randomArray);
+    }
+    public static List<Integer> generateRandomArray() {
+        List<Integer> array = new ArrayList<Integer>();
+        boolean flag = true;
+
+        while (flag) {
```
Comment (alexshamrai) [inline src/main/java/org/example/homework5/NumberGenerator.java]:
```suggestion
        while (true) {
```

File: src/main/java/org/example/homework5/NumberGenerator.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            Random random = new Random();
+            int randomNumber = random.nextInt(100) + 1;
+
+            if (array.indexOf(randomNumber) >= 0) {
+                flag = false;
```
Comment (alexshamrai) [inline src/main/java/org/example/homework5/NumberGenerator.java]:
```suggestion
                break;
```
If you use break you don't need redundant flag variable, and readability looks better

File: src/main/java/org/example/homework5/NumberGenerator.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                array.add(randomNumber);
+            }
+        }
+
+        System.out.println("Array of random generated numbers: " + array);
+
+        return array;
+    }
+
+    public static void arraySum(List<Integer> randomArray) {
```
Comment (alexshamrai) [inline src/main/java/org/example/homework5/NumberGenerator.java]: Actually using array here is also not required you can calculate the sum inside the loop

File: src/main/java/org/example/homework5/NumberGenerator.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            int randomNumber = random.nextInt(100) + 1;
+
+            if (array.indexOf(randomNumber) >= 0) {
+                flag = false;
+            } else {
+                array.add(randomNumber);
+            }
+        }
+
+        System.out.println("Array of random generated numbers: " + array);
```
Comment (alexshamrai) [inline src/main/java/org/example/homework5/NumberGenerator.java]: The task did not require ti output the array. Only total amount of numbers(like counter) and sum of generated numbers

File: src/main/java/org/example/homework5/NumberGenerator.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        boolean flag = true;
+
+        while (flag) {
+            Random random = new Random();
+            int randomNumber = random.nextInt(100) + 1;
+
+            if (array.indexOf(randomNumber) >= 0) {
+                flag = false;
+            } else {
+                array.add(randomNumber);
```
Comment (alexshamrai) [inline src/main/java/org/example/homework5/NumberGenerator.java]: We agreed in the chat that at first we should include the same number, and then exit

**Review by alexshamrai — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/homework5/EvenDigitSum.java (hunk 1 above)
Comment (anastasiya-harbatsevich) [reply src/main/java/org/example/homework5/EvenDigitSum.java]: Fixed, but left old solution as commented.

File: src/main/java/org/example/homework5/NumberGenerator.java (hunk 2 above)
Comment (anastasiya-harbatsevich) [reply src/main/java/org/example/homework5/NumberGenerator.java]: Fixed

File: src/main/java/org/example/homework5/NumberGenerator.java (hunk 3 above)
Comment (anastasiya-harbatsevich) [reply src/main/java/org/example/homework5/NumberGenerator.java]: Fixed

File: src/main/java/org/example/homework5/NumberGenerator.java (hunk 4 above)
Comment (anastasiya-harbatsevich) [reply src/main/java/org/example/homework5/NumberGenerator.java]: Fixed

File: src/main/java/org/example/homework5/NumberGenerator.java (hunk 5 above)
Comment (anastasiya-harbatsevich) [reply src/main/java/org/example/homework5/NumberGenerator.java]: Removed

File: src/main/java/org/example/homework5/NumberGenerator.java (hunk 6 above)
Comment (anastasiya-harbatsevich) [reply src/main/java/org/example/homework5/NumberGenerator.java]: Improved

**Review by alexshamrai — APPROVED** [review]

**Pull request merged**